src/extension/csw/api/src/main/java/org/geoserver/csw/store/AbstractCatalogStore.java

```diff
@@ -62,28 +61,35 @@ public CloseableIterator<String> getDomain(Name typeName, final Name attributeNa
         }
 
         // do we have such attribute?
-        final PropertyName property = rd.translateProperty(attributeName);
-        AttributeDescriptor ad = (AttributeDescriptor) property.evaluate(rd.getFeatureType());
-        if (ad == null) {
+        final List<PropertyName> properties = rd.translateProperty(attributeName);
+        boolean hasProperty = false;
+        for (PropertyName property : properties) {
+            AttributeDescriptor ad = (AttributeDescriptor) property.evaluate(rd.getFeatureType());
+            hasProperty |= ad != null;
+        }
+        if (!hasProperty) {
             return new CloseableIteratorAdapter<>(new ArrayList<String>().iterator());
         }
 
         // build the query against csw:record
         Query q = new Query(typeName.getLocalPart());
 
-        q.setProperties(Arrays.asList(translateProperty(rd, attributeName)));
+        q.setProperties(translateToPropertyNames(rd, attributeName));
 
         // collect the values without duplicates
         final Set<String> values = new HashSet<>();
         getRecords(q, Transaction.AUTO_COMMIT, rd)
                 .accepts(
                         feature -> {
-                            Property prop = (Property) property.evaluate(feature);
-                            if (prop != null) {
-                                values.add(
-                                        new String(
-                                                ((String) prop.getValue()).getBytes(ISO_8859_1),
-                                                UTF_8));
+                            for (PropertyName property : properties) {
+                                Property prop = (Property) property.evaluate(feature);
+                                if (prop != null) {
+                                    values.add(
+                                            new String(
+                                                    ((String) prop.getValue()).getBytes(ISO_8859_1),
+                                                    UTF_8));
+                                    break;
+                                }
```

Comment:
Wondering what happens if multiple xpaths for a given property happen to exist at the same time?
The documentation should probably say something about this case.